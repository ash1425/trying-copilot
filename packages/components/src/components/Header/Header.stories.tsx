import React from "react";

import { Meta, Story } from "@storybook/react";

import Header, { IHeader } from "./Header";

export default {
  component: Header,
  title: "components/Header",
  argTypes: {
    className: {
      options: "header",
      control: { type: "text" },
    },
    headerText: { options: "hello header", control: { type: "text" } },
  },
} as Meta;

const Template: Story<IHeader> = (args) => {
  const { headerText, className } = args;

  return <Header headerText={headerText} className={className} />;
};

export const Default = Template.bind({});
Default.args = { className: "header", headerText: "hello header" };
