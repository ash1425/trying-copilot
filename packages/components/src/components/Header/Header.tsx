import React from "react";

export interface IHeader {
  headerText: string;
  className: string;
}

const Header = ({ headerText, className }: IHeader) => (
  <h1 className={className}>{headerText}</h1>
);
export default Header;
