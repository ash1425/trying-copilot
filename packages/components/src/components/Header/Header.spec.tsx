import React from "react";
import { render, screen } from "@testing-library/react";
import Header from "./Header";

describe("Header", () => {
  test("renders Headercomponent", () => {
    render(<Header headerText="Test Label" className="dummy-class" />);
  });

  test("should have the text in header component", () => {
    const { container, asFragment } = render(
      <Header headerText="Test Label" className="dummy-class" />
    );
    expect(screen.getByText("Test Label")).toHaveTextContent("Test Label");
    expect(container.firstChild).toHaveClass("dummy-class");
    expect(asFragment()).toMatchSnapshot();
  });
});
