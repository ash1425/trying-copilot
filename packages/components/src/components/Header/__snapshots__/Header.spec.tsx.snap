// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should have the text in header component 1`] = `
<DocumentFragment>
  <h1
    class="dummy-class"
  >
    Test Label
  </h1>
</DocumentFragment>
`;
